dired-subtree.el

```diff
@@ -522,6 +535,69 @@ Return a string suitable for insertion in `dired' buffer."
       (dired-subtree--remove-overlays ovs)))
   (run-hooks 'dired-subtree-after-remove-hook))
 
+;;;###autoload
+(defun dired-subtree-toggle ()
+  "Insert subtree at point or remove it if it was not present."
+  (interactive)
+  (cond
+   ((dired-subtree--is-expanded-p)
+    (dired-next-line 1)
+    (dired-subtree-remove))
+   (t
+    (save-excursion
+      (dired-subtree-insert)))))
+
+(defun dired-subtree-insert-recursive (depth max-depth)
+  "Insert full subtree at point."
+  (save-excursion
+    (let ((name (dired-get-filename nil t)))
+      (when (and name (file-directory-p name)
+                 (<= depth (or max-depth depth))
+                 (or (= 1 depth)
+		     (not (string-match dired-subtree-ignored-regexp
```

Comment:
Use `string-match-p`, the non `-p` version modifies the global `match-data` which might interfere with other packages.  It is always better to use this if you don't care about the match data. (also fix the indent level :)